### PR TITLE
[CIVIS-1677] TST Optimize tests by removing marks and unneeded sleep calls

### PR DESCRIPTION
**Pull request by gkcivis**
- [x] (For Civis employees only) Reference to a relevant ticket in the pull request title
- [x] Changelog entry added to `CHANGELOG.md` at the repo's root level
- [x] Description of change in the pull request description
- [x] If applicable, unit tests have been added and/or updated
- [x] The CircleCI builds have all passed

Remove unneeded time.sleep call in test_polling.py.
Remove pytest.mark calls in test_io.py.

### CODE REVIEW

**Review by jacksonlee-civis — COMMENTED** [review]
@gkcivis This is a great start!

Would you be able to add `--durations=0` to [the pytest call in CI](https://github.com/civisanalytics/civis-python/blob/648b000fcac2ec6f91194804604397410e52fd2f/.circleci/config.yml#L54)? With this flag added, we can see how long each test takes. Currently, these are the slowest tests (showing only those that take more than one second):

```
============================== slowest durations ===============================
84.07s call     civis/tests/test_utils.py::test_retry_respect_retry_after_headers
48.08s call     civis/tests/test_utils.py::test_retry_on_retry_eligible_failures_lowercase_verbs
48.08s call     civis/tests/test_utils.py::test_retry_on_retry_eligible_failures
15.46s call     civis/tests/test_parallel.py::test_result_exception
15.14s call     civis/tests/test_parallel.py::test_result_exception_no_result
9.56s call     civis/tests/test_utils.py::test_io_retry_limit_reached
7.21s call     civis/tests/test_parallel.py::test_retries
5.73s setup    civis/tests/test_io.py::ImportTests::test_bytes_file_to_civis
5.13s call     civis/tests/test_parallel.py::test_result_eventual_failure
4.01s call     civis/tests/test_utils.py::test_retry_on_retry_eligible_post_failures
2.85s call     civis/tests/test_io.py::ImportTests::test_transfer_table
2.13s call     civis/tests/test_parallel.py::test_result_eventual_success
1.89s call     civis/tests/test_resources.py::test_duplicate_names_generated_from_api_spec
1.88s call     civis/tests/test_resources.py::test_endpoints_from_base_resources_are_available_from_client
1.75s call     civis/tests/test_civis.py::test_get_storage_host_id
1.71s call     civis/tests/test_civis.py::test_get_table_id["foo".bar]
1.57s call     civis/tests/test_futures.py::CivisFutureTests::test_set_api_result_failed
1.56s call     civis/tests/test_civis.py::test_get_table_id[foo.bar]
1.49s call     civis/tests/test_civis.py::test_get_table_id[foo."bar"]
1.49s call     civis/tests/test_civis.py::test_get_table_id["foo"."bar"]
1.48s call     civis/tests/test_client.py::ClientTests::test_feature_flags
1.48s call     civis/tests/test_futures.py::CivisFutureTests::test_polling_interval
1.47s call     civis/tests/test_futures.py::CivisFutureTests::test_check_message
1.33s call     civis/tests/test_utils.py::test_io_retry_multiple_exceptions
1.16s call     civis/tests/test_io.py::test_read_civis_sql_no_pandas
1.13s call     civis/tests/test_io.py::test_read_civis_sql_pandas
1.13s call     civis/tests/test_io.py::test_read_civis_sql_no_pandas_special_encoding_happy_path
1.12s call     civis/tests/test_io.py::test_read_civis_sql_no_pandas_special_encoding_sad_path
1.10s call     civis/tests/test_parallel.py::test_from_template_id
```

Some (but not all) of these tests should use mocking to remove the sleeping effect of `time.sleep`. For the relevant functions with sleeping due to retries or polling, we now don't want the sleeping to happen during tests (because it slows down the entire test suite) when we're just testing whether the function works and we don't care about the duration of sleeping in the test.

File: civis/tests/test_utils.py [hunk 1]
```diff
@@ -294,7 +299,7 @@ def test_no_retry_on_connection_error(mock_session):
 def test_retry_respect_retry_after_headers(mock_session):
     expected_call_count = 0
     max_calls = 3
-    retry_after = 3
+    retry_after = 1
```
Comment (gkcivis) [inline civis/tests/test_utils.py:302]: Not sure if this change was behind the intent of the ticket but by reducing this wait time the test runs much more quickly (this is the longest running test currently).

Comment (jacksonlee-civis) [reply civis/tests/test_utils.py:302]: Seems fine to me.

Please feel free to tag me again when this PR is ready for review.

Comment (gkcivis) [reply civis/tests/test_utils.py:302]: Trying to shorten a couple more

File: .circleci/config.yml [hunk 2]
```diff
@@ -51,7 +51,7 @@ jobs:
           # Avoid being able to import civis by relative import.
           # Test code by importing the *installed* civis in site-packages.
           command: |
-              pytest -vv --cov=project/civis --junitxml=/tmp/testxml/report.xml project/civis
+              pytest -vv --cov=project/civis --junitxml=/tmp/testxml/report.xml project/civis --durations=0
```
Comment (gkcivis) [inline .circleci/config.yml:54]: Once everything else looks good should I revert this change?

**Review by jacksonlee-civis — DISMISSED** [review]
LGTM!

🚀  Awesome work! The step to run the test suite on CI went from 5 minutes to just 1 minute.

**Review by jacksonlee-civis — APPROVED** [review]
Still LGTM!
